lib/create-request-body.ts

```diff
@@ -1,18 +1,43 @@
 import { TraeSettings } from '../src/types';
+import { isValidBody, isHeaders } from './guards';
 
-function isJSON(config: TraeSettings) {
-  const headers = config.headers;
-  // @ts-ignore
-  const key = Object.keys(headers).find(
-    // @ts-ignore
-    (header: any) => header.toLowerCase() === 'content-type',
+function isJSON({ headers }: TraeSettings) {
```

Comment:
I don’t like the way we are handling headers. It seems strange having 3 posible data structures to represent the same data.
I am not sure why we need the Array.isArray method, were are we using header as arrays?
Any ways, is not a comment to update your implementation, we can handle this in another PR. Maybe we can only use the Headers interface of the Fetch API everywhere somehow.

Reply:
Sorry, I should've explained why I did this.

It's not "the way we are handling headers". Fetch supports all of those types (array of pairs `string[][]`, the `Headers` object and records `Record<string, string>`).

That is the reason why it was not type checking. There were cases not covered.

If we want to support all potential ways a user might provide headers to a fetch request we need to do something along the lines of what I did (maybe there's a better implementation but the three types need to be covered).

Reply:
We can indeed use the Fetch `Headers` since the constructor supports the other two ways headers might be provided

https://developer.mozilla.org/en-US/docs/Web/API/Headers (see the examples section).

This is actually quite cool, because we don't need to then iterate the array or the object, just do `headers.get() ...`

Reply:
Great! Indeed looks way nicer! 💯